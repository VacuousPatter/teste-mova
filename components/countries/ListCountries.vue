<template>
  <v-row class="countries__wrapper">
    <v-col
      v-for="(country,i) in countriesPageList"
      :key="i"
      cols="12"
      md="6"
      lg="4"
      class="countries"
    >
      <CountryCard :country="country" />
    </v-col>
    <v-col v-show="pagesCount>1" cols="12">
      <v-row justify="center">
        <v-col cols="12" sm="8" md="6" lg="4">
          <v-pagination v-model="page" class="paginacao" :length="pagesCount" />
        </v-col>
      </v-row>
    </v-col>
  </v-row>
</template>

<script>
import CountryCard from './CountryCard.vue'

export default {
  components: { CountryCard },

  props: {
    countries: {
      type: Array,
      required: true
    },
    itemsPerPage: {
      type: Number,
      default: 25
    }
  },

  data () {
    return {
      page: 1,
      pagesCount: Math.ceil(this.countries.length / this.itemsPerPage)
    }
  },

  computed: {
    countriesPageList () {
      return this.countries.slice(this.page * this.itemsPerPage - this.itemsPerPage, this.page * this.itemsPerPage)
    }
  },

  watch: {
    countries (val) {
      this.pagesCount = Math.ceil(val.length / this.itemsPerPage)
    }
  }
}
</script>

<style scoped>
.countries__wrapper{
  overflow: hidden;
}
</style>

<style>
.countries__country{
  position: relative;
  margin: 0px 21px;
}

@media (max-width:600px) {
  .countries__country{
    margin: 0;
  }
}
</style>
