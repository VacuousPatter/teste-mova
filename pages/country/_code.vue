<template>
  <v-container>
    <v-row v-if="country">
      <v-col cols="12">
        <v-row>
          <v-col cols="12" md="3">
            <v-img class="mx-auto" max-width="400px" :src="country.flags.png || country.coatOfArms.png" :aspect-ratio="16/9" />
          </v-col>
          <v-col cols="">
            <v-list class="country__details">
              <v-list-item class="details__detail">
                <v-list-item-title>Nome Local: {{ localCountryName }}</v-list-item-title>
              </v-list-item>
              <v-list-item class="details__detail">
                <v-list-item-title>Nome Internacional: {{ country.name.common }}</v-list-item-title>
              </v-list-item>
              <v-list-item class="details__detail">
                <v-list-item-title>Capital: {{ country.capital[0] }}</v-list-item-title>
              </v-list-item>
              <v-list-item class="details__detail">
                <v-list-item-title>
                  Região: <nuxt-link
                    :to="{
                      path:'/',
                      query:{
                        region: country.region
                      }
                    }"
                  >
                    {{ country.region }}
                  </nuxt-link>
                </v-list-item-title>
              </v-list-item>
              <v-list-item class="details__detail">
                <v-list-item-title>Sub-região: {{ country.subregion }}</v-list-item-title>
              </v-list-item>
              <v-list-item class="details__detail">
                <v-list-item-title>População: {{ formatPopulationNumber(country.population) }}</v-list-item-title>
              </v-list-item>
              <v-list-item class="details__detail">
                <v-list-item-title>Linguas: {{ countryLanguages }}</v-list-item-title>
              </v-list-item>
            </v-list>
          </v-col>
        </v-row>
        <v-row>
          <v-col cols="12">
            <v-card tile flat>
              <v-card-title>Países Vizinhos:</v-card-title>

              <ListCountries v-if="borderCountries.length>0" :countries="borderCountries" :items-per-page="12" />
              <v-alert v-else text>
                Esse país não possui nenhum país vizinho.
              </v-alert>
            </v-card>
          </v-col>
        </v-row>
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
import ListCountries from '../../components/countries/ListCountries.vue'

export default {
  components: { ListCountries },

  data () {
    return {
      country: null,
      borderCountries: []
    }
  },

  computed: {
    countryLanguages () {
      if (!this.country) { return null }
      const arr = Object.values(this.country.languages)
      const strLanguages = arr.reduce((p, lang, index, array) => {
        if (index === 0) {
          p += lang
        } else if (index + 1 === array.length) {
          p += ` e ${lang}`
        } else {
          p += `, ${lang}`
        }

        return p
      }, '')

      return strLanguages
    },
    localCountryName () {
      return this.country.name.nativeName && Object.values(this.country.name.nativeName)[0] ? Object.values(this.country.name.nativeName)[0].common : this.country.name.common
    }
  },

  mounted () {
    this.getCountry()
  },

  methods: {
    getCountry () {
      this.$axios.$get(`https://restcountries.com/v3.1/alpha/${this.$route.params.code}`)
        .then((countries) => {
          this.country = countries[0]
          this.getCountryBorderCountries()
        })
    },
    getCountryBorderCountries () {
      this.$axios.$get('https://restcountries.com/v3.1/alpha', {
        params: {
          codes: this.country.borders.join(',')
        }
      })
        .then((countries) => {
          this.borderCountries = countries
        })
    },
    formatPopulationNumber (number) {
      const numberArr = number.toString().split('')
      const resto = numberArr.length % 3
      if (resto === 0) {
        return numberArr.slice(0, 3).join('') + ' ' + this.getNumberCardinalName(number)
      } else if (resto === 1) {
        return numberArr.slice(0, 2).join('') + ' ' + this.getNumberCardinalName(number)
      } else if (resto === 2) {
        return numberArr.slice(0, 1).join('') + ' ' + this.getNumberCardinalName(number)
      }
    },
    getNumberCardinalName (number) {
      const numberSize = number.toString().length
      if (numberSize >= 4 && numberSize <= 6) {
        return 'Mil'
      } else if (numberSize >= 7 && numberSize <= 9) {
        return 'Milhões'
      } else if (numberSize >= 10 && numberSize <= 12) {
        return 'Bilhões'
      }
      return ''
    }
  }
}
</script>

<style scoped>
.details__detail{
  color: #454545 !important;
}
</style>
