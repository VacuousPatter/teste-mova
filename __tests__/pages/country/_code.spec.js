import { mount, createLocalVue, RouterLinkStub } from '@vue/test-utils'
import Vuetify from 'vuetify'
import _code from '@/pages/country/_code'
import ListCountries from '@/components/countries/ListCountries'
const data = require('./data.json')

describe('/pages/country/_code.vue', () => {
  describe('Deve mostrar o país', () => {
    let wrapper, axiosMock

    beforeEach(async () => {
      const localVue = createLocalVue()

      axiosMock = {
        $get: jest.fn((url, options) => {
          return new Promise((resolve) => {
            if (url === 'https://restcountries.com/v3.1/alpha/BRA') {
              resolve(data.country)
            } else if (url === 'https://restcountries.com/v3.1/alpha') {
              if (options.params.codes === 'ARG,BOL,COL,GUF,GUY,PRY,PER,SUR,URY,VEN') {
                resolve(data.borderCountries)
              }
            } else {
              resolve([])
            }
          })
        })
      }

      wrapper = mount(_code, {
        localVue,
        vuetify: new Vuetify(),
        mocks: {
          $axios: axiosMock,
          $route: {
            params: {
              code: 'BRA'
            }
          }
        },
        stubs: {
          NuxtLink: RouterLinkStub
        }
      })

      await wrapper.vm.$nextTick()
    })

    test('mostrar informações na tela', () => {
      const nomeLocal = wrapper.find('.details__detail--nome-local')
      expect(nomeLocal.exists()).toBeTruthy()
      expect(nomeLocal.isVisible()).toBeTruthy()
      expect(nomeLocal.element.textContent).toEqual('Nome Local: Brasil')

      const nomeInternacional = wrapper.find('.details__detail--nome-internacional')
      expect(nomeInternacional.exists()).toBeTruthy()
      expect(nomeInternacional.isVisible()).toBeTruthy()
      expect(nomeInternacional.element.textContent).toEqual('Nome Internacional: Brazil')

      const nomeCapital = wrapper.find('.details__detail--nome-capital')
      expect(nomeCapital.exists()).toBeTruthy()
      expect(nomeCapital.isVisible()).toBeTruthy()
      expect(nomeCapital.element.textContent).toEqual('Capital: Brasília')

      const nomeRegiao = wrapper.find('.details__detail--nome-regiao-link')
      expect(nomeRegiao.exists()).toBeTruthy()
      expect(nomeRegiao.isVisible()).toBeTruthy()
      expect(nomeRegiao.element.textContent.trim()).toEqual('Americas')

      const nomeSubRegiao = wrapper.find('.details__detail--nome-subregiao')
      expect(nomeSubRegiao.exists()).toBeTruthy()
      expect(nomeSubRegiao.isVisible()).toBeTruthy()
      expect(nomeSubRegiao.element.textContent).toEqual('Sub-região: South America')

      const populacao = wrapper.find('.details__detail--populacao')
      expect(populacao.exists()).toBeTruthy()
      expect(populacao.isVisible()).toBeTruthy()
      expect(populacao.element.textContent).toEqual('População: 212 Milhões')

      const linguas = wrapper.find('.details__detail--linguas')
      expect(linguas.exists()).toBeTruthy()
      expect(linguas.isVisible()).toBeTruthy()
      expect(linguas.element.textContent).toEqual('Linguas: Portuguese')

      const ListCountriesWrapper = wrapper.findComponent(ListCountries)
      expect(ListCountriesWrapper.exists()).toBeTruthy()
      expect(ListCountriesWrapper.isVisible()).toBeTruthy()

      expect(axiosMock.$get).toHaveBeenCalledTimes(2)

      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('Deve mostrar o país e mensagem sem paises vizinhos', () => {
    let wrapper, axiosMock

    beforeEach(async () => {
      const localVue = createLocalVue()

      axiosMock = {
        $get: jest.fn((url, options) => {
          return new Promise((resolve) => {
            if (url === 'https://restcountries.com/v3.1/alpha/BRA') {
              resolve(data.country)
            } else if (url === 'https://restcountries.com/v3.1/alpha') {
              if (options.params.codes === 'ARG,BOL,COL,GUF,GUY,PRY,PER,SUR,URY,VEN') {
                resolve([])
              }
            } else {
              resolve([])
            }
          })
        })
      }

      wrapper = mount(_code, {
        localVue,
        mocks: {
          $axios: axiosMock,
          $route: {
            params: {
              code: 'BRA'
            }
          }
        },
        stubs: {
          NuxtLink: RouterLinkStub
        }
      })

      await wrapper.vm.$nextTick()
    })

    test('mostrar informações na tela', () => {
      const nomeLocal = wrapper.find('.details__detail--nome-local')
      expect(nomeLocal.exists()).toBeTruthy()
      expect(nomeLocal.isVisible()).toBeTruthy()
      expect(nomeLocal.element.textContent).toEqual('Nome Local: Brasil')

      const nomeInternacional = wrapper.find('.details__detail--nome-internacional')
      expect(nomeInternacional.exists()).toBeTruthy()
      expect(nomeInternacional.isVisible()).toBeTruthy()
      expect(nomeInternacional.element.textContent).toEqual('Nome Internacional: Brazil')

      const nomeCapital = wrapper.find('.details__detail--nome-capital')
      expect(nomeCapital.exists()).toBeTruthy()
      expect(nomeCapital.isVisible()).toBeTruthy()
      expect(nomeCapital.element.textContent).toEqual('Capital: Brasília')

      const nomeRegiao = wrapper.find('.details__detail--nome-regiao-link')
      expect(nomeRegiao.exists()).toBeTruthy()
      expect(nomeRegiao.isVisible()).toBeTruthy()
      expect(nomeRegiao.element.textContent.trim()).toEqual('Americas')

      const nomeSubRegiao = wrapper.find('.details__detail--nome-subregiao')
      expect(nomeSubRegiao.exists()).toBeTruthy()
      expect(nomeSubRegiao.isVisible()).toBeTruthy()
      expect(nomeSubRegiao.element.textContent).toEqual('Sub-região: South America')

      const populacao = wrapper.find('.details__detail--populacao')
      expect(populacao.exists()).toBeTruthy()
      expect(populacao.isVisible()).toBeTruthy()
      expect(populacao.element.textContent).toEqual('População: 212 Milhões')

      const linguas = wrapper.find('.details__detail--linguas')
      expect(linguas.exists()).toBeTruthy()
      expect(linguas.isVisible()).toBeTruthy()
      expect(linguas.element.textContent).toEqual('Linguas: Portuguese')

      const ListCountriesWrapper = wrapper.findComponent(ListCountries)
      expect(ListCountriesWrapper.exists()).toBeFalsy()

      const MensagemSemPaisesVizinho = wrapper.find('.mensagem-sem-pais-vizinho')
      expect(MensagemSemPaisesVizinho.exists()).toBeTruthy()

      expect(axiosMock.$get).toHaveBeenCalledTimes(2)

      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('Não mostrar país', () => {
    let wrapper, axiosMock

    beforeEach(async () => {
      const localVue = createLocalVue()

      axiosMock = {
        $get: jest.fn((url, options) => {
          return new Promise((resolve) => {
            if (url === 'https://restcountries.com/v3.1/alpha/BRA') {
              resolve([])
            } else if (url === 'https://restcountries.com/v3.1/alpha') {
              if (options.params.codes === 'ARG,BOL,COL,GUF,GUY,PRY,PER,SUR,URY,VEN') {
                resolve([])
              }
            } else {
              resolve([])
            }
          })
        })
      }

      wrapper = mount(_code, {
        localVue,
        mocks: {
          $axios: axiosMock,
          $route: {
            params: {
              code: 'BRA'
            }
          }
        },
        stubs: {
          NuxtLink: RouterLinkStub
        }
      })

      await wrapper.vm.$nextTick()
    })

    test('mostrar informações na tela', () => {
      const countryWrapper = wrapper.find('.country-wrapper')
      expect(countryWrapper.exists()).toBeFalsy()

      const avisoPaisNaoEncontrado = wrapper.find('.erro-pais-404')
      expect(avisoPaisNaoEncontrado.exists()).toBeTruthy()

      expect(axiosMock.$get).toHaveBeenCalledTimes(1)

      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
