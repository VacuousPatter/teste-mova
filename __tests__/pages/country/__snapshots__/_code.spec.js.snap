// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/pages/country/_code.vue Deve mostrar o país e mensagem sem paises vizinhos mostrar informações na tela 1`] = `
"<div class=\\"container\\">
  <div class=\\"row country-wrapper\\">
    <div class=\\"col col-12\\">
      <div class=\\"row\\">
        <div class=\\"col-md-3 col-12\\">
          <div class=\\"v-image v-responsive mx-auto theme--light\\" style=\\"max-width: 400px;\\">
            <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
            <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
              <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
            </transition-stub>
            <div class=\\"v-responsive__content\\"></div>
          </div>
        </div>
        <div class=\\"col col-true\\">
          <div role=\\"list\\" class=\\"v-list country__details v-sheet theme--light\\">
            <div tabindex=\\"-1\\" role=\\"listitem\\" class=\\"details__detail details__detail--nome-local v-list-item theme--light\\">
              <div class=\\"v-list-item__title\\">Nome Local: Brasil</div>
            </div>
            <div tabindex=\\"-1\\" role=\\"listitem\\" class=\\"details__detail details__detail--nome-internacional v-list-item theme--light\\">
              <div class=\\"v-list-item__title\\">Nome Internacional: Brazil</div>
            </div>
            <div tabindex=\\"-1\\" role=\\"listitem\\" class=\\"details__detail details__detail--nome-capital v-list-item theme--light\\">
              <div class=\\"v-list-item__title\\">Capital: Brasília</div>
            </div>
            <div tabindex=\\"-1\\" role=\\"listitem\\" class=\\"details__detail details__detail--nome-regiao v-list-item theme--light\\">
              <div class=\\"v-list-item__title\\">
                Região:
                <a class=\\"details__detail--nome-regiao-link\\">
                  Americas
                </a></div>
            </div>
            <div tabindex=\\"-1\\" role=\\"listitem\\" class=\\"details__detail details__detail--nome-subregiao v-list-item theme--light\\">
              <div class=\\"v-list-item__title\\">Sub-região: South America</div>
            </div>
            <div tabindex=\\"-1\\" role=\\"listitem\\" class=\\"details__detail details__detail--populacao v-list-item theme--light\\">
              <div class=\\"v-list-item__title\\">População: 212 Milhões</div>
            </div>
            <div tabindex=\\"-1\\" role=\\"listitem\\" class=\\"details__detail details__detail--linguas v-list-item theme--light\\">
              <div class=\\"v-list-item__title\\">Linguas: Portuguese</div>
            </div>
          </div>
        </div>
      </div>
      <div class=\\"row\\">
        <div class=\\"col col-12\\">
          <div class=\\"v-card v-card--flat v-sheet theme--light rounded-0\\">
            <div class=\\"v-card__title\\">Países Vizinhos:</div>
            <div role=\\"alert\\" class=\\"v-alert mensagem-sem-pais-vizinho v-sheet theme--light v-alert--text\\">
              <div class=\\"v-alert__wrapper\\">
                <div class=\\"v-alert__content\\">
                  Esse país não possui nenhum país vizinho.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`/pages/country/_code.vue Deve mostrar o país mostrar informações na tela 1`] = `
"<div class=\\"container\\">
  <div class=\\"row country-wrapper\\">
    <div class=\\"col col-12\\">
      <div class=\\"row\\">
        <div class=\\"col-md-3 col-12\\">
          <div class=\\"v-image v-responsive mx-auto theme--light\\" style=\\"max-width: 400px;\\">
            <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
            <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
              <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
            </transition-stub>
            <div class=\\"v-responsive__content\\"></div>
          </div>
        </div>
        <div class=\\"col col-true\\">
          <div role=\\"list\\" class=\\"v-list country__details v-sheet theme--light\\">
            <div tabindex=\\"-1\\" role=\\"listitem\\" class=\\"details__detail details__detail--nome-local v-list-item theme--light\\">
              <div class=\\"v-list-item__title\\">Nome Local: Brasil</div>
            </div>
            <div tabindex=\\"-1\\" role=\\"listitem\\" class=\\"details__detail details__detail--nome-internacional v-list-item theme--light\\">
              <div class=\\"v-list-item__title\\">Nome Internacional: Brazil</div>
            </div>
            <div tabindex=\\"-1\\" role=\\"listitem\\" class=\\"details__detail details__detail--nome-capital v-list-item theme--light\\">
              <div class=\\"v-list-item__title\\">Capital: Brasília</div>
            </div>
            <div tabindex=\\"-1\\" role=\\"listitem\\" class=\\"details__detail details__detail--nome-regiao v-list-item theme--light\\">
              <div class=\\"v-list-item__title\\">
                Região:
                <a class=\\"details__detail--nome-regiao-link\\">
                  Americas
                </a></div>
            </div>
            <div tabindex=\\"-1\\" role=\\"listitem\\" class=\\"details__detail details__detail--nome-subregiao v-list-item theme--light\\">
              <div class=\\"v-list-item__title\\">Sub-região: South America</div>
            </div>
            <div tabindex=\\"-1\\" role=\\"listitem\\" class=\\"details__detail details__detail--populacao v-list-item theme--light\\">
              <div class=\\"v-list-item__title\\">População: 212 Milhões</div>
            </div>
            <div tabindex=\\"-1\\" role=\\"listitem\\" class=\\"details__detail details__detail--linguas v-list-item theme--light\\">
              <div class=\\"v-list-item__title\\">Linguas: Portuguese</div>
            </div>
          </div>
        </div>
      </div>
      <div class=\\"row\\">
        <div class=\\"col col-12\\">
          <div class=\\"v-card v-card--flat v-sheet theme--light rounded-0\\">
            <div class=\\"v-card__title\\">Países Vizinhos:</div>
            <div class=\\"row countries__wrapper\\">
              <div class=\\"countries col-md-6 col-lg-4 col-12\\"><a tabindex=\\"0\\" class=\\"countries__country v-card v-card--link v-sheet theme--light\\">
                  <div class=\\"v-responsive\\">
                    <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                    <div class=\\"v-responsive__content\\">
                      <div class=\\"v-image v-responsive theme--light\\">
                        <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                        <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
                          <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
                        </transition-stub>
                        <div class=\\"v-responsive__content\\"></div>
                      </div>
                    </div>
                  </div>
                  <transition-stub name=\\"expand-transition\\">
                    <!---->
                  </transition-stub>
                </a></div>
              <div class=\\"countries col-md-6 col-lg-4 col-12\\"><a tabindex=\\"0\\" class=\\"countries__country v-card v-card--link v-sheet theme--light\\">
                  <div class=\\"v-responsive\\">
                    <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                    <div class=\\"v-responsive__content\\">
                      <div class=\\"v-image v-responsive theme--light\\">
                        <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                        <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
                          <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
                        </transition-stub>
                        <div class=\\"v-responsive__content\\"></div>
                      </div>
                    </div>
                  </div>
                  <transition-stub name=\\"expand-transition\\">
                    <!---->
                  </transition-stub>
                </a></div>
              <div class=\\"countries col-md-6 col-lg-4 col-12\\"><a tabindex=\\"0\\" class=\\"countries__country v-card v-card--link v-sheet theme--light\\">
                  <div class=\\"v-responsive\\">
                    <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                    <div class=\\"v-responsive__content\\">
                      <div class=\\"v-image v-responsive theme--light\\">
                        <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                        <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
                          <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
                        </transition-stub>
                        <div class=\\"v-responsive__content\\"></div>
                      </div>
                    </div>
                  </div>
                  <transition-stub name=\\"expand-transition\\">
                    <!---->
                  </transition-stub>
                </a></div>
              <div class=\\"countries col-md-6 col-lg-4 col-12\\"><a tabindex=\\"0\\" class=\\"countries__country v-card v-card--link v-sheet theme--light\\">
                  <div class=\\"v-responsive\\">
                    <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                    <div class=\\"v-responsive__content\\">
                      <div class=\\"v-image v-responsive theme--light\\">
                        <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                        <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
                          <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
                        </transition-stub>
                        <div class=\\"v-responsive__content\\"></div>
                      </div>
                    </div>
                  </div>
                  <transition-stub name=\\"expand-transition\\">
                    <!---->
                  </transition-stub>
                </a></div>
              <div class=\\"countries col-md-6 col-lg-4 col-12\\"><a tabindex=\\"0\\" class=\\"countries__country v-card v-card--link v-sheet theme--light\\">
                  <div class=\\"v-responsive\\">
                    <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                    <div class=\\"v-responsive__content\\">
                      <div class=\\"v-image v-responsive theme--light\\">
                        <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                        <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
                          <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
                        </transition-stub>
                        <div class=\\"v-responsive__content\\"></div>
                      </div>
                    </div>
                  </div>
                  <transition-stub name=\\"expand-transition\\">
                    <!---->
                  </transition-stub>
                </a></div>
              <div class=\\"countries col-md-6 col-lg-4 col-12\\"><a tabindex=\\"0\\" class=\\"countries__country v-card v-card--link v-sheet theme--light\\">
                  <div class=\\"v-responsive\\">
                    <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                    <div class=\\"v-responsive__content\\">
                      <div class=\\"v-image v-responsive theme--light\\">
                        <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                        <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
                          <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
                        </transition-stub>
                        <div class=\\"v-responsive__content\\"></div>
                      </div>
                    </div>
                  </div>
                  <transition-stub name=\\"expand-transition\\">
                    <!---->
                  </transition-stub>
                </a></div>
              <div class=\\"countries col-md-6 col-lg-4 col-12\\"><a tabindex=\\"0\\" class=\\"countries__country v-card v-card--link v-sheet theme--light\\">
                  <div class=\\"v-responsive\\">
                    <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                    <div class=\\"v-responsive__content\\">
                      <div class=\\"v-image v-responsive theme--light\\">
                        <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                        <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
                          <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
                        </transition-stub>
                        <div class=\\"v-responsive__content\\"></div>
                      </div>
                    </div>
                  </div>
                  <transition-stub name=\\"expand-transition\\">
                    <!---->
                  </transition-stub>
                </a></div>
              <div class=\\"countries col-md-6 col-lg-4 col-12\\"><a tabindex=\\"0\\" class=\\"countries__country v-card v-card--link v-sheet theme--light\\">
                  <div class=\\"v-responsive\\">
                    <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                    <div class=\\"v-responsive__content\\">
                      <div class=\\"v-image v-responsive theme--light\\">
                        <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                        <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
                          <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
                        </transition-stub>
                        <div class=\\"v-responsive__content\\"></div>
                      </div>
                    </div>
                  </div>
                  <transition-stub name=\\"expand-transition\\">
                    <!---->
                  </transition-stub>
                </a></div>
              <div class=\\"countries col-md-6 col-lg-4 col-12\\"><a tabindex=\\"0\\" class=\\"countries__country v-card v-card--link v-sheet theme--light\\">
                  <div class=\\"v-responsive\\">
                    <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                    <div class=\\"v-responsive__content\\">
                      <div class=\\"v-image v-responsive theme--light\\">
                        <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                        <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
                          <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
                        </transition-stub>
                        <div class=\\"v-responsive__content\\"></div>
                      </div>
                    </div>
                  </div>
                  <transition-stub name=\\"expand-transition\\">
                    <!---->
                  </transition-stub>
                </a></div>
              <div class=\\"countries col-md-6 col-lg-4 col-12\\"><a tabindex=\\"0\\" class=\\"countries__country v-card v-card--link v-sheet theme--light\\">
                  <div class=\\"v-responsive\\">
                    <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                    <div class=\\"v-responsive__content\\">
                      <div class=\\"v-image v-responsive theme--light\\">
                        <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
                        <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
                          <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
                        </transition-stub>
                        <div class=\\"v-responsive__content\\"></div>
                      </div>
                    </div>
                  </div>
                  <transition-stub name=\\"expand-transition\\">
                    <!---->
                  </transition-stub>
                </a></div>
              <div class=\\"col col-12\\" style=\\"display: none;\\">
                <div class=\\"row justify-center\\">
                  <div class=\\"col-sm-8 col-md-6 col-lg-4 col-12\\">
                    <nav role=\\"navigation\\" aria-label=\\"Pagination Navigation\\" class=\\"paginacao\\">
                      <ul class=\\"v-pagination theme--light\\">
                        <li><button disabled=\\"disabled\\" type=\\"button\\" aria-label=\\"Previous page\\" class=\\"v-pagination__navigation v-pagination__navigation--disabled\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-chevron-left theme--light\\"></i></button></li>
                        <li><button type=\\"button\\" aria-current=\\"true\\" aria-label=\\"Current Page, Page 1\\" class=\\"v-pagination__item v-pagination__item--active primary\\">1</button></li>
                        <li><button disabled=\\"disabled\\" type=\\"button\\" aria-label=\\"Next page\\" class=\\"v-pagination__navigation v-pagination__navigation--disabled\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-chevron-right theme--light\\"></i></button></li>
                      </ul>
                    </nav>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`/pages/country/_code.vue Não mostrar país mostrar informações na tela 1`] = `
"<div class=\\"container\\">
  <div class=\\"row erro-pais-404\\">
    <div class=\\"col col-12\\">
      <div role=\\"alert\\" class=\\"v-alert v-sheet theme--light v-alert--text\\">
        <div class=\\"v-alert__wrapper\\">
          <div class=\\"v-alert__content\\">
            País não encontrado no banco de dados!
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
